### PR TITLE
Kia and Hyundai Support

### PR DESCRIPTION
**Pull request by emmertex**
Working on:
Kia Sorento
Kia Stinger
Hyundai Elantra
Hyundai Genesis

Tuning is WIP: https://docs.google.com/spreadsheets/d/1ZfqAHr1FkchRechv8Pjci9ls7dQ68zuDSPlpvYr7Bz4/edit#gid=0

Safety is WIP

### CODE REVIEW

File: selfdrive/car/hyundai/carstate.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+    # Gear Selecton - This should be compatible with all Kia/Hyundai with Auto's
+    if cp.vl["CLU15"]["CF_Clu_InhibitD"] == 1:
+      self.gear_shifter = "drive"
+    elif cp.vl["CLU15"]["CF_Clu_InhibitN"] == 1:
+      self.gear_shifter = "neutral"
+    elif cp.vl["CLU15"]["CF_Clu_InhibitP"] == 1:
+      self.gear_shifter = "park"
+    elif cp.vl["CLU15"]["CF_Clu_InhibitR"] == 1:
+      self.gear_shifter = "reverse"
```
Comment (rbiasini) [inline selfdrive/car/hyundai/carstate.py]: are the cruise buttons (set/resume) on can?

Comment (emmertex) [reply selfdrive/car/hyundai/carstate.py]: They are, yes.

File: selfdrive/car/hyundai/interface.py [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+    ret.minEnableSpeed = -1.
+
+    centerToRear = ret.wheelbase - ret.centerToFront
+    # TODO: get actual value, for now starting with reasonable value for
+    # civic and scaling by mass and wheelbase
+    ret.rotationalInertia = rotationalInertia * ret.mass * ret.wheelbase**2
+
+    # TODO: start from empirically derived lateral slip stiffness for the civic and scale by
+    # mass and CG position, so all cars will have approximately similar dyn behaviors
+    ret.tireStiffnessFront = tireStiffnessFront * ret.mass * (centerToRear / ret.wheelbase)
```
Comment (rbiasini) [inline selfdrive/car/hyundai/interface.py]: @emmertex this is an important bug: seem my port for correct dimensional scaling.
As is, car should strongly underestimate the needed steering angle to make a turn

**Comment by rbiasini** [comment]
obsoleted by https://github.com/commaai/openpilot/pull/346

**Pull request closed without merging**